src/hip_hcc.cpp

```diff
@@ -1409,9 +1410,38 @@ void ihipInit()
     tprintf(DB_SYNC, "pid=%u %-30s g_numLogicalThreads=%u\n", getpid(), "<ihipInit>", g_numLogicalThreads);
 }
 
+hipError_t ihipStreamSynchronize(hipStream_t stream)
+{
+    hipError_t e = hipSuccess;
 
+    if (stream == hipStreamNull) {
+        ihipCtx_t *ctx = ihipGetTlsDefaultCtx();
+        ctx->locked_syncDefaultStream(true/*waitOnSelf*/, true/*syncToHost*/);
+    } else {
+        // note this does not synchornize with the NULL stream:
+        stream->locked_wait();
+        e = hipSuccess;
+    }
 
+    return e;
```

Comment:
It looks like e is always hipSuccess.

Reply:
ihipStreamSynchronize only returns when the stream is synchronized. So it will return hipSuccess. In To be consistent with the rest of the code, instead of directly using return hipSuccess, we initialize e to hipSuccess and then return e.

Reply:
It always returns hipSuccess.